### DISCUSSION TITLE
GotFocus event on TableViewColumn

### DISCUSSION THREAD

**Discussion by JLBLU** [w-ahmad/WinUI.TableView#86] [Feedback Wanted]
Hi Ahmad

Is it possible to add the GotFocus event on TableViewColum ?
Thanks

**Comment by w-ahmad** [comment]
What are you trying to achieve with this event?

**Reply by JLBLU** [reply]
Hi Ahamad, 
Finally, scrap this request.
I wanted to use this event to try to resolve the problem described in the message : _Edit TableViewTextColumn #73_.
Because I had some difficulty managing List entry. I had a command bar button to enable/disable the entry in list and in some cases, tableview column not always getting the focus : if several clicks on same column, or click in first row. Not always reproducible ... Moreover several click on some row select/unselect the row, this is not the expected (for me) behaviour.
But now, I decided to handle my TableView always in readonly mode and I added a new command bar button to open a kind of detail page in order to edit my row. Then, all works fine with some little tips. But I'm not sure I'm doing everything right ...

               <controlsTV:TableView
                       x:Name="tvDkSettingsDocFolders" 
                       Background="LightGray" 
                       IsDoubleTapEnabled="True"
                       AutoGenerateColumns="False"
                       SelectionMode="Single"
                       SelectionUnit="Row"
                       SelectedItem="{Binding SelectedDocFolderItem, Mode=TwoWay}"
                       SelectionChanged="tvDkSettingsDocFolders_SelectionChanged"
                       IsReadOnlyChanged="tvDkSettingsDocFolders_IsReadOnlyChanged"
                       AllowFocusWhenDisabled="True"
                       GotFocus="tvDkSettingsDocFolders_GotFocus">
                    
        private TableViewRow FindRowUnderMouse(DependencyObject dp)
        {
            TableViewRow dg = dp.GetParentOfType<TableViewRow>();
            return dg;
        }

        private void EnforceRowSelection(DependencyObject dp)
        {
            if (dp != null)
            {
                if (dp is TextBox)
                {
                    TableViewRow dgr = FindRowUnderMouse(dp);
                    bool doSelect = false;

                    if (dgr != null)
                    {
                        RowClicked = dgr.Content as DkSettingsFolderViewModel; 
                        RowClickedIndex = dgr.Index;

                        if (RowClicked != null)
                        {
                            doSelect = LastRowClicked == null ? true :
                                            RowClicked == LastRowClicked ? false : true;

                            if (doSelect)
                            {
                                SettingsVM.SelectedFolderItem = RowClicked;
                                SelectListViewItem(RowClickedIndex);

                            }
                            LastRowClicked = RowClicked;
                        }
                    }
                }
            }
        }

        private void SelectListViewItem(int idx)
        {
            ListViewItem lvi = tvDkSettingsDocFolders.ContainerFromIndex(idx) as ListViewItem;
            if (lvi != null)
            {
                lvi.IsSelected = true;
                lvi.Focus(FocusState.Programmatic);
            }
        }

        private void tvDkSettingsFolders_GotFocus(object sender, RoutedEventArgs e)
        {
            DependencyObject dp = (DependencyObject)e.OriginalSource;
            EnforceRowSelection(dp);
        }

// 
Combo-Box #71 is more annoying (but I found a workaround)
Thanks
Good luck for the future
